### PR TITLE
Rewrite ipc to grpc

### PR DESCRIPTION
**Pull request by pettermk**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,19 +2,26 @@
 FROM debian:bullseye as cppbuild
 RUN apt-get update && apt-get install -y zlib1g-dev libgflags-dev libboost-graph-dev libtsan0 \
       libboost-serialization-dev libboost-test-dev libboost-iostreams-dev libpython3-dev libzmq3-dev \
-      libhiredis-dev pybind11-dev wget build-essential git python3 python3-distutils file 
+      libhiredis-dev pybind11-dev wget build-essential git python3 python3-distutils file \
+      apt-transport-https gnupg python autoconf libtool pkg-config python3-venv curl zip unzip tar \
+      python3-pip
```
Comment (pettermk) [inline Dockerfile:7]: Check if can be removed

File: grpc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+target_include_directories(hierarchy_service_grpc_proto
+    PRIVATE ${CMAKE_SOURCE_DIR}
+)
+
+target_link_libraries(hierarchy_service_grpc_proto
+    PUBLIC gRPC::grpc++_reflection
+    PUBLIC gRPC::grpc++
+    PUBLIC protobuf::libprotobuf
+    tag_hierarchy
```
Comment (pettermk) [inline grpc/CMakeLists.txt:41]: Can it be removed?

File: grpc/test_type_conversion.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        auto remote_node = Node();
+        TypeConversion::ConvertToRemoteNode(local, &remote_node);
+        auto roundtripped_local = TypeConversion::ConvertToLocalNode(remote_node);
+        BOOST_TEST(roundtripped_local == local);
+    };
+
+    BOOST_AUTO_TEST_CASE( test_convert_to_remote_node )
+    {
+
+    };
```
Comment (pettermk) [inline grpc/test_type_conversion.cpp:35]: Delete

File: vcpkg_settings/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    "name": "ioc-hierarchy-service",
+    "version": "1.5.0",
+    "dependencies": [
+        {
+            "name": "grpc",
+            "version==": "1.33.1"
+        }
+    ],
+    "builtin-baseline": "977c4b612d72b02eda9693d9207a5007f055d721"
+}
```
Comment (pettermk) [inline vcpkg_settings/vcpkg.json:11]: Delete

**Comment by equinor-ruaj** [comment]
cherry-picked the two first commits:
- Refactor filters into separate file for reuse
- Remove opencensus library 

In PR: https://github.com/equinor/ioc-hierarchy-service/pull/33

Check if rebase or something is needed before merging this

**Review by erikmaland — APPROVED** [review]
From the perspective of my limited grpc knowledge, it looks good.

**Review by erikmaland — APPROVED** [review]

**Pull request merged**